File: 0001_0100/0029/0029.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    p = [int(x) for x in f.readline().split()]
    s = 0 if n == 1 else abs(p[1] - p[0])
    prev = 0
    for i in range(2, n):
        t = s
        s = min(s + abs(p[i] - p[i - 1]), prev + 3 * abs(p[i] - p[i - 2]))
        prev = t
    q.write(str(s))

File: 0001_0100/0029/0029.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func abs(x int32) int32 {
	y := x >> 31
	return (x ^ y) - y
}

func min(a, b int32) int32 {
	if a < b {
		return a
	}
	return b
}

func main() {
	q, _ := os.Create("output.txt")
	content, _ := ioutil.ReadFile("input.txt")
	d := strings.Fields(string(content))
	n, _ := strconv.Atoi(d[0])
	p := make([]int, n)
	for i := 0; i < n; i++ {
		p[i], _ = strconv.Atoi(d[i+1])
	}
	var s, prev int32
	if n > 1 {
		s = abs(int32(p[1] - p[0]))
	}
	for i := 2; i < n; i++ {
		s, prev = min(s+abs(int32(p[i]-p[i-1])), prev+3*abs(int32(p[i]-p[i-2]))), s
	}
	fmt.Fprint(q, s)
}
